javascript/barcharts.js

```diff
@@ -0,0 +1,289 @@
+        .text(yAxisLabel);
+
+      svg.selectAll(".bar")
+        .data(data)
+        .enter()
+        .append("rect")
+        .attr("width", 0)
+        .attr("x", 0)
+        // .attr("x", 800) // for fancy animation
+        .transition().duration(bars.animation.duration)
+        .delay(function(d, i) {
+          return i * bars.animation.delay
+        })
+        .attr("class", "bar")
+        // .attr("x", 0) // for fancy animation
+        .attr("width", function(d) {
+          return ((x(d[xColumn])) / totalWidth) * 100 + "%";
+          var test = 0;
+        })
+        .attr("y", function(d) {
+          return (y(d[yColumn]) + bars.padding);
+        })
+        .attr("height", bars.height)
+        .filter(function(d) {
+          return d[yColumn] == 0;
+        }).remove();
+
+      // Bar labels
+      svg.selectAll(".text")
+        .data(data)
+        .enter()
+        .append("text")
+        .attr("class", "barLabel")
+        .style("fill", "white") // fill: white; is not a good approach coz bckgr can be not white //NEED FIX
+        .transition()
+        .duration(300)
+        .delay(function(d, i) {
+          return i * bars.animation.delay + bars.animation.duration - 400
+        })
+        .text(function(d, i) {
+          return formatPercent(d[xColumn])
+        })
+        .attr('x', function(d) {
+          return ((x(d[xColumn]) + format.labels.position.x) / totalWidth) * 100 + "%";
+        })
+        .attr('y', function(d) {
+          return y(d[yColumn]) + bars.height / 2 + format.labels.position.y + bars.padding;
+        })
+        .style("fill", format.labels.color)
+        .style("display", function(d) {
+          if (!x(d[xColumn])) {
+            return "none"
+          }
+        })
+        .style("text-anchor", format.labels.position.anchor)
+        .filter(function(d) {
+          return d[yColumn] == 0;
+        }).remove();
+
+      // Bar titles
+      svg.selectAll(".text")
+        .data(data)
+        .enter()
+        .append('text')
+        .attr('class', 'barTitle')
+        .text(function(d) {
+          return d[yColumn]
+        })
+        .attr('x', function(d) {
+          return format.titles.position.x;
+        })
+        .attr('y', function(d) {
+          return y(d[yColumn]) + bars.height / 2 + format.titles.position.y + bars.padding;
+        })
+        .style("fill", format.titles.color)
+        .style("text-anchor", format.titles.position.anchor)
+        .filter(function(d) {
+          return d[yColumn] == 0;
+        }).remove();
+
+    });
+
+    function type(d) {
+      // d.Age = +d.Age; //is it necessary to convert or not????????
+      return d;
+    }
+  }
+
+  render();
+
+  return {
+    //API
+  };
+
+});
+
+//  barCharts() prepares all variables to be passed into drawBarCharts()
+var barCharts = (function(topic, format, leftMargin, rightMargin) {
+
+  //data for bar charts
+  var dataBC = './data/2016-New-Coder-Survey-Data-Summary.csv';
+
```

Comment:
I'm using one file to draw all the bar charts. It kind of slowing down the process because every time for rendering we are downloading the summary file.

Approaches:
- make 20 different small .csv files for each bar chart (is that okey?)
- manually input the data into the code (not very flexible)
- try to download data once and then store it in the code

Which approach is better?